docs/csharp/language-reference/statements/exception-handling-statements.md

```diff
@@ -0,0 +1,141 @@
+---
+title: "Exception-handling statements - throw and try"
+description: "Use the C# throw statement to signal an occurrence of an exception. Use the C# try statements to catch and process exceptions occurred in a block of code."
+ms.date: 04/21/2023
+f1_keywords:
+  - "throw_CSharpKeyword"
+  - "try_CSharpKeyword"
+  - "catch_CSharpKeyword"
+  - "finally_CSharpKeyword"
+  - "catch-finally_CSharpKeyword"
+helpviewer_keywords:
+  - "throw statement [C#]"
+  - "throw expression [C#]"
+  - "throw keyword [C#]"
+  - "try statement [C#]"
+  - "try-catch statement [C#]"
+  - "try-finally statement [C#]"
+  - "try-catch-finally statement [C#]"
+  - "try keyword [C#]"
+  - "catch keyword [C#]"
+  - "finally keyword [C#]"
+  - "when keyword [C#]"
+---
+# Exception-handling statements - `throw`, `try-catch`, `try-finally`, and `try-catch-finally`
+
+You use the `throw` and `try` statements to work with exceptions. Use the [`throw` statement](#the-throw-statement) to throw an exception. Use the [`try` statement](#the-try-statement) to catch and handle exceptions that might occur during execution of a code block.
+
+## The `throw` statement
+
+The `throw` statement throws an exception:
+
+:::code language="csharp" source="snippets/exception-handling-statements/Program.cs" id="Throw":::
+
+In a `throw e;` statement, the result of expression `e` must be implicitly convertible to <xref:System.Exception?displayProperty=nameWithType>. You can use the built-in exception classes, for example, <xref:System.ArgumentOutOfRangeException> or <xref:System.InvalidOperationException>. You can also define your own exception classes that derive from <xref:System.Exception?displayProperty=nameWithType>. For more information, see [Creating and throwing exceptions](../../fundamentals/exceptions/creating-and-throwing-exceptions.md).
```

Comment:
What do you think about mentioning helpers like "ThrowIfNull" here:

```suggestion
In a `throw e;` statement, the result of expression `e` must be implicitly convertible to <xref:System.Exception?displayProperty=nameWithType>. You can use the built-in exception classes, for example, <xref:System.ArgumentOutOfRangeException> or <xref:System.InvalidOperationException>. You can also use helper methods like <xref:System.ArgumentNullException.ThrowIfNull%2A?displayProperty=nameWithType> which throw exceptions based on conditions. You can also define your own exception classes that derive from <xref:System.Exception?displayProperty=nameWithType>. For more information, see [Creating and throwing exceptions](../../fundamentals/exceptions/creating-and-throwing-exceptions.md).
```

Reply:
I like that idea, thanks!